### PR TITLE
Add TLS for SAPM and SignalFx receiver

### PR DESCRIPTION
**Pull request by ccaraman**
**Description:** Adds TLS for both SAPM and SignalFx Receiver.

**Testing:** Added tests to create a TLS connection to the receiver and manually tested using Otel Collector and Smart Agent.

**Documentation:** Expanded readme's to add information about TLS.

### CODE REVIEW

**Comment by ccaraman** [comment]
cc @brianashby-sfx for doc changes.

**Comment by bogdandrutu** [comment]
@ccaraman I think we should share the config from core, see:
https://github.com/open-telemetry/opentelemetry-collector/pull/927

Do you think that is possible?

**Comment by jrcamp** [comment]
We were also trying to do this in https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/198

**Comment by ccaraman** [comment]
@bogdandrutu 
> @ccaraman I think we should share the config from core, see:
> [open-telemetry/opentelemetry-collector#927](https://github.com/open-telemetry/opentelemetry-collector/pull/927)
> 
> Do you think that is possible?

Yes, that is totally possible. I didn't see that PR so I didn't know that was coming in the pipeline. I'll take a look 

@jrcamp - should we then wait on https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/198 and get the first mentioned pr to work for HTTP as well?

**Comment by jrcamp** [comment]
I think the upstream issue linked above will supersede most of #198. (Though we're doing some other common config structs for HTTP that will be built on top of TLSConfig but shouldn't block your work here.)

**Comment by ccaraman** [comment]
@bogdandrutu and @jrcamp - since this is using the common receiver settings (for http server) - i don't think this blocks this pr getting reviewed. There is testing logic around https connection so that can be reviewed before the refactor of the tls configuration.

File: receiver/sapmreceiver/trace_receiver_test.go [hunk 1]
```diff
@@ -232,13 +278,22 @@ func TestReception(t *testing.T) {
...
+			require.NoError(t, receivedErr, "should not have failed to start trace reception")
+			require.False(t, receivedError)
 			t.Log("Trace Reception Started")
 
 			t.Log("Sending Sapm Request")
 			var resp *http.Response
-			resp, err = sendSapm(tt.args.config.Endpoint, tt.args.sapm, tt.args.zipped)
-			assert.NoError(t, err, fmt.Sprintf("should not have failed when sending sapm %v", resp))
+			resp, err = sendSapm(tt.args.config.Endpoint, tt.args.sapm, tt.args.zipped, tt.args.useTLS)
+			require.NoError(t, err, fmt.Sprintf("should not have failed when sending sapm %v", err))
```
Comment (jrcamp) [inline receiver/sapmreceiver/trace_receiver_test.go:295]:
```suggestion
			require.NoErrorf(t, err, "should not have failed when sending sapm %v", err)
```

File: receiver/signalfxreceiver/receiver_test.go [hunk 2]
```diff
@@ -333,6 +319,120 @@ func Test_sfxReceiver_handleReq(t *testing.T) {
...
+	client := &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{
+				RootCAs: caCertPool,
+			},
+		},
+	}
+
+	resp, err := client.Do(req)
+	require.NoError(t, err, fmt.Sprintf("should not have failed when sending to signalFx receiver %v", err))
```
Comment (jrcamp) [inline receiver/signalfxreceiver/receiver_test.go:411]:
```suggestion
	require.NoErrorf(t, err, "should not have failed when sending to signalFx receiver %v", err)
```

File: receiver/signalfxreceiver/receiver_test.go [hunk 3]
```diff
@@ -333,6 +319,120 @@ func Test_sfxReceiver_handleReq(t *testing.T) {
...
+
+	t.Log("Sending SignalFx metric data Request")
+
+	body, err := proto.Marshal(buildSFxMsg(&msec, 13, 3))
+	require.NoError(t, err, fmt.Sprintf("failed to marshal SFx message: %v", err))
+
+	url := fmt.Sprintf("https://%s/v2/datapoint", addr)
+
+	req, err := http.NewRequest("POST", url, bytes.NewReader(body))
+	require.NoError(t, err, fmt.Sprintf("should have no errors with new request: %v", err))
```
Comment (jrcamp) [inline receiver/signalfxreceiver/receiver_test.go:394]:
```suggestion
	require.NoErrorf(t, err, "should have no errors with new request: %v", err)
```

File: receiver/signalfxreceiver/receiver_test.go [hunk 4]
```diff
@@ -333,6 +319,120 @@ func Test_sfxReceiver_handleReq(t *testing.T) {
...
+	req.Header.Set("Content-Type", "application/x-protobuf")
+
+	caCert, err := ioutil.ReadFile("./testdata/testcert.crt")
+	require.NoError(t, err, fmt.Sprintf("failed to load certificate: %v", err))
```
Comment (jrcamp) [inline receiver/signalfxreceiver/receiver_test.go:398]:
```suggestion
	require.NoErrorf(t, err, "failed to load certificate: %v", err)
```

**Review by jrcamp — APPROVED** [review]
lgtm just a couple suggestions

**Review by jrcamp — APPROVED** [review]

**Review by tigrannajaryan — APPROVED** [review]

**Pull request merged**